app/src/main/java/com/example/cinemapp/ui/main/home/HomeFragment.kt

```diff
@@ -70,9 +70,24 @@ class HomeFragment : Fragment() {
         }
     }
 
+    private fun setupOnScrollListener() {
+        binding.rvMovieList.addOnScrollListener(
+            object : RecyclerView.OnScrollListener() {
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+                    if (binding.rvMovieList.isEndOfScroll()) {
+                        viewModel.setupLoading()
```

Comment:
Fragment knows the order of methods in which to call ViewModel. Encapsulate these two functions into one called `onScrollEnd`